lib/bgpstream_elem.c

```diff
@@ -327,6 +330,18 @@ char *bgpstream_elem_custom_snprintf(char *buf, size_t len,
     if (B_FULL)
       return NULL;
 
+#ifdef WITH_RPKI
+    /* RPKI validation */
+    if(elem->annotations.rpki_active){
+      char result[B_REMAIN];
+      if(bgpstream_rpki_validate(elem, result, sizeof(result))){
```

Comment:
I added the following comment, I hope it is more understandable? 
```
/* If the RPKI parameter was set, the corresponding input was valid and the 
    configure was completely set up -> the elem can be validated by RPKI */
```
If ```bgpstream_rpki_validate``` returns 0 the BGP output buffer is not extended by the RPKI validation string (```result```). Therefore the output of the element contains BGP informations only.

Reply:
The comment helps!

If the validation failed (e.g. ROA invalid), will the validation string (`result`) show some error message?

Reply:
Can you also add some comment [here](https://github.com/CAIDA/libbgpstream/pull/26/files/99e682f7170c8786ce42386c39664c9729c41564#diff-c604ba7a7a21a3e41d9b5e5a5d5a9060R201) to explain the `if` statements? 

Seems to me that `rpki_validate` returns 0 if success (assuming `val_rst==1` means success), is it correct?

Reply:
The validation results in three correct outputs: valid, invalid and notfound. In these cases ```result``` is set with the corresponding output. If the validation fails as a result of an error ```result``` is not set and the standard BGPReader-output will be displayed. The ROAFetchlib prints a debug error message instead in order to avoid error messages in the output.

Reply:
Yes it is correct. The meaning of the binary output is different. I added the following comment.
```
  /* Validate the BGP elem only if the origin ASN is a simple ASN value
     (i.e. not a set). If the validation function of the ROAFetchlib
     returns 0 a valid result (val_rst = 1) is available */
```